test/Makefile

```diff
@@ -0,0 +1,68 @@
+LIBDIR=../lib
+GTEST_MODULE=$(LIBDIR)/googletest
+GTEST_DIR=$(LIBDIR)/googletest/googletest
+GTEST_SOURCEDIR=$(GTEST_DIR)/src
+GTEST_HEADERS=$(GTEST_DIR)/include/gtest/*.h \
+              $(GTEST_DIR)/include/gtest/internal/*.h
+GTEST_SRCS_=$(GTEST_SOURCEDIR)/*.cc $(GTEST_SOURCEDIR)/*.h $(GTEST_HEADERS)
+
+# TODO: Factor out the common variables.
+CPP_NETLIB_DIR=$(LIBDIR)/cpp-netlib
+YAML_CPP_DIR=$(LIBDIR)/yaml-cpp
+
+CPPFLAGS+=-isystem $(GTEST_DIR)/include
+CXXFLAGS=-std=c++11 -g -pthread
+LDFLAGS=$(CPPFLAGS)
+LDLIBS=-lpthread
```

Comment:
[Optional] This may actually be implied by `-pthread`, but it's harmless to keep as-is.

Reply:
This is required.

Reply:
This may not have been required had we set `LDFLAGS=$(CXXFLAGS)`, but it's not a big deal either way.